src/datasets/exceptions.py

```diff
@@ -83,3 +84,107 @@ def from_cast_error(
             explanation_message += f"\n\nThis happened while the {builder_name} dataset builder was generating data using\n\n{', '.join(formatted_tracked_gen_kwargs)}"
         help_message = "\n\nPlease either edit the data files to have matching columns, or separate them into different configurations (see docs at https://hf.co/docs/hub/datasets-manual-configuration#multiple-configurations)"
         return cls("An error occurred while generating the dataset" + explanation_message + help_message)
+
+
+@deprecated("Use 'ChecksumVerificationError' instead.")
+class ChecksumVerificationException(Exception):
+    """Exceptions during checksums verifications of downloaded files.
+
+    <Deprecated version="2.20.0">
+
+    Use `ChecksumVerificationError` instead.
+
+    </Deprecated>
+    """
+
+
+class ChecksumVerificationError(DatasetsError, ChecksumVerificationException):
+    """Error raised during checksums verifications of downloaded files."""
+
+
+@deprecated("Use 'UnexpectedDownloadedFileError' instead.")
+class UnexpectedDownloadedFile(ChecksumVerificationException):
+    """Some downloaded files were not expected.
+
+    <Deprecated version="2.20.0">
+
+    Use `UnexpectedDownloadedFileError` instead.
+
+    </Deprecated>
+    """
+
+
+class UnexpectedDownloadedFileError(ChecksumVerificationError, UnexpectedDownloadedFile):
+    """Some downloaded files were not expected."""
+
+
+@deprecated("Use 'ExpectedMoreDownloadedFilesError' instead.")
+class ExpectedMoreDownloadedFiles(ChecksumVerificationException):
+    """Some files were supposed to be downloaded but were not.
+
+    <Deprecated version="2.20.0">
+
+    Use `ExpectedMoreDownloadedFilesError` instead.
+
+    </Deprecated>
+    """
+
+
+class ExpectedMoreDownloadedFilesError(ChecksumVerificationError, ExpectedMoreDownloadedFiles):
+    """Some files were supposed to be downloaded but were not."""
+
+
+class NonMatchingChecksumError(ChecksumVerificationError):
+    """The downloaded file checksum don't match the expected checksum."""
+
+
+@deprecated("Use 'SplitsVerificationError' instead.")
+class SplitsVerificationException(Exception):
+    """Exceptions during splits verifications.
+
+    <Deprecated version="2.20.0">
+
+    Use `SplitsVerificationError` instead.
+
+    </Deprecated>
+    """
+
+
+class SplitsVerificationError(DatasetsError, SplitsVerificationException):
```

Comment:
is it also marked as deprecated since it inherits from SplitsVerificationException ?

Reply:
I have renamed `SplitsVerificationException` to `SplitsVerificationError` and deprecated the former. The new `SplitsVerificationError` is not deprecated. 

It inherits from `SplitsVerificationException` for backward compatibility: if a user is catching `SplitsVerificationException`, now it will also catch `SplitsVerificationError`.

Reply:
@lhoestq I fixed it.